test/TestTools.jl

```diff
@@ -40,7 +40,8 @@ function SphericalShellSpace(;
         boundary_names = (:bottom, :top),
     )
     vertmesh = ClimaCore.Meshes.IntervalMesh(vertdomain; nelems = zelem)
-    vert_center_space = ClimaCore.Spaces.CenterFiniteDifferenceSpace(vertmesh)
+    vert_center_space =
+        ClimaCore.Spaces.CenterFiniteDifferenceSpace(context, vertmesh)
```

Comment:
With something like
```suggestion
if pkgversion(ClimaCore) < v"0.15" 
        vert_center_space =
        ClimaCore.Spaces.CenterFiniteDifferenceSpace(context)
  else
          vert_center_space =
        ClimaCore.Spaces.CenterFiniteDifferenceSpace(context, vertmesh)
end
```
we can keep compatibility with previous versions of ClimaCore and merge this now.

Reply:
So, I decided to deprecate the device-less methods (https://github.com/CliMA/ClimaCore.jl/pull/1825), so I think we can actually just incrementally make changes, and then make a breaking release once all of the upstream packages have conformed.

Reply:
The only place where the change is required is in the test suite, so I think it would be an overkill to bump the compat for ClimaCore just for this.

Reply:
It's now a non-breaking version, though

Reply:
I don't understand: CI fails in this PR, so I take that this change is not compatible with `ClimaCore` `0.14.9` (and previous versions). I think we should still maintain compatibility with previous versions when it is easy.

Reply:
Yeah, we need to release a new version and update to the newly supported feature, it's still (for now) backwards compatible.